### PR TITLE
Kubelet register a hostname Provider

### PR DESCRIPTION
**Pull request by JulienBalestra**
### What does this PR do?

The kubelet register a Hostname Provider function.
This function allows to get the `spec.nodeName` of any Pod running on the node.

It cannot work if the PodList is empty.

### CODE REVIEW

File: pkg/util/kubernetes/kubelet/kubelet.go [hunk 1]
```diff
@@ -78,9 +78,17 @@ func GetKubeUtil() (*KubeUtil, error) {
...
-// GetNodeInfo returns the IP address and the hostname of the node where
-// this pod is running.
-func (ku *KubeUtil) GetNodeInfo() (ip, name string, err error) {
+// HostnameProvider kubelet implementation for the hostname provider
+func HostnameProvider(hostName string) (string, error) {
+	ku, err := GetKubeUtil()
+	if err != nil {
+		return "", err
+	}
+	return ku.GetHostname()
```
Comment (xvello) [inline pkg/util/kubernetes/kubelet/kubelet.go]: let's be explicit we're returning `ku.GetHostname(), nil`

File: releasenotes/notes/kubelet-hostname-provider-c3755d746fea6122.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+features:
+  - |
+    The kubelet register a Hostname Provider function.
```
Comment (xvello) [inline releasenotes/notes/kubelet-hostname-provider-c3755d746fea6122.yaml]: release notes are directly concatenated to the changelog published to users, could you rephrase it as a user-facing feature instead of a dev PoV? Something in the style of "added ability to retrieve the hostname from the kubernetes APIs"?

**Review by xvello — COMMENTED** [review]
LGTM! 2 nitpicks

File: pkg/util/kubernetes/kubelet/kubelet.go (hunk 1 above)
Comment (JulienBalestra) [reply pkg/util/kubernetes/kubelet/kubelet.go]: Oh I thought `ku.GetHostname()` already returned the tuple string, error. Do you think it’s worth putting as a comment ?
```go
func (ku *KubeUtil) GetHostname() (string, error)
```
I mirrored the style we used [here](https://github.com/DataDog/datadog-agent/blob/master/pkg/util/docker/global.go#L73).

File: releasenotes/notes/kubelet-hostname-provider-c3755d746fea6122.yaml (hunk 2 above)
Comment (JulienBalestra) [reply releasenotes/notes/kubelet-hostname-provider-c3755d746fea6122.yaml]: Got it. I made the changes.

**Comment by JulienBalestra** [comment]
@xvello BTW, I voluntary dismissed [isKubernetes](https://github.com/DataDog/datadog-agent/pull/1131/files#diff-634b0b54c640c77a2bde1eaf176061b3L27) because I was thinking of a use case running inside a docker container not launched by a kubelet.

But it's still possible to have a kubelet on the node/host.

File: pkg/util/kubernetes/kubelet/kubelet.go (hunk 1 above)
Comment (xvello) [reply pkg/util/kubernetes/kubelet/kubelet.go]: indeed, my bad

**Comment by xvello** [comment]
When in doubt, I think it's better to replicate the agent5 logic, and test for `KUBERNETES=yes` (we set it in our example daemonset)

This provider is only needed if:
  - not running on the host (else the system hostname is deemed OK)
  - docker not available
  - kubelet configured and available

**Review by xvello — DISMISSED** [review]
👍

File: pkg/util/kubernetes/kubelet/kubelet.go [hunk 3]
```diff
@@ -290,13 +292,13 @@ func (ku *KubeUtil) init() error {
 	// HTTPS first
 	_, errHTTPS = c.Get(fmt.Sprintf("https://%s:%d/", ku.kubeletHost, config.Datadog.GetInt("kubernetes_https_kubelet_port")))
 	if errHTTPS != nil {
-		log.Debugf("cannot connect: %s", errHTTPS)
+		log.Debugf("Cannot connect: %s, try trough http", errHTTPS)
```
Comment (xvello) [inline pkg/util/kubernetes/kubelet/kubelet.go]: `s/try/trying/`

**Review by xvello — APPROVED** [review]
🍔
